CSharp/Library/Microsoft.Bot.Connector.Shared/ServiceProvider.cs

```diff
@@ -99,5 +107,49 @@ private static void ThrowOnNullInstance()
 
             return service;
         }
+
+        /// <summary>
+        /// Tries to auto register the ASP.NET implementation of Connector for backward compatibility.
+        /// </summary>
+        private static void TryAutoRegisterForBackwardCompatibility()
+        {
+            const string AspNetBotConnectorAssemblyName = "Microsoft.Bot.Connector";
+            const string AspNetBotConnectorServiceProviderFullQualifiedName = "Microsoft.Bot.Connector.BotServiceProvider";
+
+            Assembly connectorAssembly = null;
+
+            try
+            {
+                var connectorAssemblyName = new AssemblyName(AspNetBotConnectorAssemblyName);
+                connectorAssembly = Assembly.Load(connectorAssemblyName);
+            }
+            catch (Exception)
+            {
+                // assembly not available
+                // cannot log, because we don't have service provider to get logger from
+            }
+
+            if (connectorAssembly != null)
+            {
+                Type botServiceProviderType = connectorAssembly.DefinedTypes
+                        .FirstOrDefault(t => t.FullName.Equals(AspNetBotConnectorServiceProviderFullQualifiedName))
```

Comment:
The "t.FullName" here is for the type full name (i.e. namespace + class name), not the assembly's.